### PR TITLE
VBM-75: Feature/personal info with backend 

### PR DESCRIPTION
**Pull request by VladimirBludov**

### CODE REVIEW

File: packages/mobile/src/screens/personal-info/personal-info.tsx [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
 
 const PersonalInfoScreen: FC = () => {
-  const onSubmit = (): void => {
-    // TODO: handle submit
-  };
+  const dispatch = useAppDispatch();
+  const personalInfo = useAppSelector(selectPersonalInfo);
+
+  useFocusEffect(
+    React.useCallback(() => {
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/personal-info/personal-info.tsx:14]: import this from `hooks`

File: packages/mobile/src/store/auth/selectors.ts [hunk 2]
```diff
@@ -2,16 +2,23 @@ import { RootState } from '~/common/types/types';
 
 const selectCurrentUser = (state: RootState) => state.auth.user;
 
+const selectUserId = (state: RootState) => state.auth.user?.id;
+
 const selectUserPhone = (state: RootState) => state.auth.user?.phone;
 
 const selectPhoneVerified = (state: RootState) =>
   state.auth.user?.phoneVerified;
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/store/auth/selectors.ts:10]: Such precise selectors doesn't make much sense. We can replace it with single `selectCurrentUser`

File: packages/mobile/src/store/personal-info/actions.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+const updatePersonalInfo = createAsyncThunk<
+  SaveUserProfileDto,
+  SaveUserProfileDto,
+  AsyncThunkConfig
+>(ActionType.UPDATE_PERSONAL_INFO, async (payload, { extra }) => {
+  const { personalInfoApi } = extra;
+  const parsedPayload = updatePersonalInfoParser(payload);
+  const response = await personalInfoApi.updatePersonalInfo(parsedPayload);
+  const personalInfo = personalInfoParser(response);
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/store/personal-info/actions.ts:30]: These data manipulations are not obvious. Consider removing them at all or keep is as helper and apply before dispatching the action

**Review by AleksandrTermenzhy — APPROVED** [review]

**Review by dolphin323 — APPROVED** [review]
Merge after fixing upper comments

File: packages/mobile/src/screens/personal-info/personal-info.tsx (hunk 1 above)
Comment (VladimirBludov) [reply packages/mobile/src/screens/personal-info/personal-info.tsx:14]: corrected

File: packages/mobile/src/store/auth/selectors.ts (hunk 2 above)
Comment (VladimirBludov) [reply packages/mobile/src/store/auth/selectors.ts:10]: corrected

**Review by natata7 — APPROVED** [review]

File: packages/mobile/src/store/personal-info/actions.ts (hunk 3 above)
Comment (VladimirBludov) [reply packages/mobile/src/store/personal-info/actions.ts:30]: corrected

**Pull request merged**